### PR TITLE
Add integration tests

### PR DESCRIPTION
**Pull request by qtomlinson**
Add initial integration tests for harvest and definition comparison. More test cases will be added in the future to cover all the component types.

An example run can be found at https://github.com/qtomlinson/operations/actions/runs/7924449799

### CODE REVIEW

File: tools/integration/test/testConfig.js
Comment (lumaxis) [inline tools/integration/test/testConfig.js:33]: Why are these commented out?

File: tools/integration/test/serviceTest.js
Comment (lumaxis) [inline tools/integration/test/serviceTest.js:5]: You could also be consistent with the other imports here, if you want 🙂

```suggestion
const { expect } = require('chai')
```

File: tools/integration/test/lib/harvesterTest.js
Comment (lumaxis) [inline tools/integration/test/lib/harvesterTest.js:4]: Same here.

```suggestion
const { expect } = require('chai')
```

File: tools/integration/test/lib/pollerTest.js
Comment (lumaxis) [inline tools/integration/test/lib/pollerTest.js:4]:
```suggestion
const { expect } = require('chai')
```

File: tools/integration/clearlydefined_api_test_samples_insomnia.json
Comment (lumaxis) [inline tools/integration/clearlydefined_api_test_samples_insomnia.json:1]: Is this file intentionally committed? If so, maybe a small comment in the Readme on how to use it would be nice.

File: tools/integration/test/testConfig.js
Comment (qtomlinson) [reply tools/integration/test/testConfig.js:33]: These are the type of components that yet need good test cases.

File: tools/integration/clearlydefined_api_test_samples_insomnia.json
Comment (elrayle) [reply tools/integration/clearlydefined_api_test_samples_insomnia.json:1]: For context, I imported this file into PostMan and was able to use it to verify that the production service was running  as expected after updating the deploy process.  So there primary use is for ad hoc testing the API.

It would be helpful to have documentation that...
*  identifies how each request is confirming the function of the API
* the expected results of each request
* what it might look like if it fails

I would suggest adding the health check URL as the most basic and first check to see if the service is even up.

**Review by elrayle — COMMENTED** [review]
This looks good to me.  I defer to @lumaxis review for the code completeness and correctness.  I had one question about adding documentation for how to use the insomnia API file for ad hoc testing.  It is not critical that it be part of this PR.  A follow-on PR would be fine.

Thanks for adding these.  Having integration tests are going to help a lot with confidence that the full process is working as expected.

File: tools/integration/clearlydefined_api_test_samples_insomnia.json
Comment (qtomlinson) [reply tools/integration/clearlydefined_api_test_samples_insomnia.json:1]: Will add some explaining to Readme.  The collection is primarily organized by end points (definitions, harvest, and notices), which has detail documentation at https://api.clearlydefined.io/api-docs/#/. The search and sort are quite specific for fixing continuation token purposes, which will be removed.

**Comment by qtomlinson** [comment]
@lumaxis @elrayle ready for review.

File: tools/integration/lib/fetch.js
Comment (lumaxis) [inline tools/integration/lib/fetch.js:5]: Fwiw, we could totally decide to run these tests with Node 21 right now and use native fetch 😇

File: tools/integration/lib/harvester.js
Comment (lumaxis) [inline tools/integration/lib/harvester.js:10]: What I find slightly confusing here is that this uses the schema version not the tool version. Not sure how to make this clearer and avoid confusion for future maintainers though 😅

File: tools/integration/package.json
Comment (lumaxis) [inline tools/integration/package.json:18]: One thought I had today: I think with out current usage, we don't really need a third-party dependency for assertions and could just rely on the native `assert`. Nothing to necessarily address right now but maybe in the future.

**Review by lumaxis — APPROVED** [review]
💪🏼

File: tools/integration/package.json
Comment (qtomlinson) [reply tools/integration/package.json:18]: True.  That will reduce the maintenance effort in the future.

File: tools/integration/lib/harvester.js
Comment (qtomlinson) [reply tools/integration/lib/harvester.js:10]: I can also add some comments here and testConfig.js to clarify this.

File: tools/integration/lib/fetch.js
Comment (qtomlinson) [reply tools/integration/lib/fetch.js:5]: hmm... Node LTS is preferred.  It is also intentionally kept in sync with setup-node action in service, crawler for easier maintenance.

File: tools/integration/lib/harvester.js
Comment (qtomlinson) [reply tools/integration/lib/harvester.js:10]: Will address this as a separate PR.

**Pull request merged**
